esmvaltool/utils/cmorizers/obs/cmorize_obs_Landschutzer2014.py

```diff
@@ -0,0 +1,117 @@
+"""
+# #############################################################################
+# ESMValTool CMORizer for WOA data
+# #############################################################################
+#
+# Tier
+#    Tier 2: other freely-available dataset.
+#
+# Source
+#    ftp://ftp.nodc.noaa.gov/nodc/archive/arc0105/0160558/1.1/data/0-data/
+#
+# Last access
+#    20190227
+#
+# Download and processing instructions
+#    Download the following file in ${RAWOBS}/Tier2/Landschutzer2014:
+#     pco2_1998-2011_ETH_SOM-FFN_CDIAC_G05.nc
+#      
+#
+# Modification history
+#    20190227-A_lova_to: written.
+#
+# #############################################################################
+"""
+
+import datetime
+import logging
+import os
+import glob
+
+import iris
+import netCDF4 as nc
+from cf_units import Unit
+import numpy as np
+
+from .utilities import (_add_metadata,
+                        _convert_timeunits,
+                        _fix_coords,
+                        _read_cmor_config,
+                        _roll_cube_data,
+                        _save_variable)
+
+logger = logging.getLogger(__name__)
+
+# read in CMOR configuration
+cfg = _read_cmor_config('Landschutzer2014.yml')
+proj = cfg['proj']
+timestamp = datetime.datetime.utcnow()
+timestamp_format = "%Y-%m-%d %H:%M:%S"
+now_time = timestamp.strftime(timestamp_format)
+proj['metadata_attributes']['CMORcreated'] = now_time
+VAR_TO_CMOR = cfg['VAR_TO_CMOR']
+VAR_TO_FILENAME = cfg['VAR_TO_FILENAME']
+STANDARD_NAMES = cfg['STANDARD_NAMES']
+LONG_NAMES = cfg['LONG_NAMES']
+NC4_ZIP = cfg['NC4_ZIP']
+# 
+ALLVARS=list(VAR_TO_CMOR.keys())
+
+
+def _fix_metadata(cube, var):
+    """Fix all aspects of metadata for different vars."""
+    logger.info("Fixing units...")
+    if var in ['fgco2',] :
+        cube.units = Unit('kg m-2 s-1')
+    if var in ['spco2','dpco2' ] :
+        cube.units = Unit('Pa')
+    return cube
+
+
+def _fix_data(cube, var):
+    """Specific data fixes for different variables."""
+    logger.info("Fixing data ...")
+    # fix for missing value definition 
+    cube.data=np.ma.masked_values(cube.data, cube.data.fill_value)
```

Comment:
I don't understand why are you rebuilding the mask - if the the cube data has a `fill_value` attr it also means it has a mask with elements masked on that value; are you trying to check if the cube has a mask?

Reply:
the problem stems in the original dataset that come with bad missing values definiton (1.e+20 instead of 1.e+20f) so the cube fails to assign the mask to data when loading.

`cube.data(1.e+20, 1.e+20, 1.e+20, 300., ... 1.e+20, mask=False,  fill_value=1.e+20)`

To correct this problem, I put in the data function the mask operation to fix it and I also added the assignment of the missing value when saving the data with iris (following this old issue https://github.com/ESMValGroup/ESMValTool/issues/302#issuecomment-381099386)

Reply:
yes, at save point it's a good idea, cheers, I have omitted that, but has to be done with a check if data is masked or not. But at load point it's strange that the cube is not loaded with a mask already (probably the `var: _FillValue = 1.e+20f` and `var: missing_value = 1.e+20f` are missing from the cube attributes); may be related to an iris issue I opened a few days ago; anyway, you can build the mask in-place `cube.data = np.ma.masked_array(cube.data, mask=(cube.data == 1.e20))` or the way you did it, the only problem is the way you doing it you also need to check that the fill_value is actually the cmor one (1e20) since iris2 loads cubes with default fill_value of 1e36

Reply:
missing values of this dataset are a mess, here below an example
` fgco2_smoothed:FillValue = 1.e+20 ; fgco2_smoothed:missing_value = 1.e+20 ;`

Note that in this case the cube is loaded anyway with a correct fill_value that is 1.e+20 (but without applying the mask) and iris throws the  the following warning 
`iris/fileformats/netcdf.py:395: UserWarning: WARNING: missing_value not used since it
cannot be safely cast to variable data type`

Reply:
@valeriupredoi do you agree in closing this one?